spmd/__init__.py

```diff
@@ -1,106 +1,96 @@
 # Copyright (c) Meta Platforms, Inc. and affiliates
-from typing import Sequence, Optional, cast
+from typing import Optional, Callable, Tuple, Union
 import torch
 import torch.nn as nn
-from spmd.tensor import DTensor, Placement, Shard, Replicate
+from spmd.tensor import distribute_tensor
+from spmd.tensor.api import DTensor
 from spmd.tensor.device_mesh import get_global_device_mesh, DeviceMesh
+from spmd.tensor.placement_types import Replicate, Shard
 
-torch.__future__.set_overwrite_module_params_on_conversion(True)
 
-
-def distribute_tensor(
-    tensor: torch.Tensor,
+def distribute_module(
+    module: nn.Module,
     device_mesh: Optional[DeviceMesh] = None,
-    placements: Optional[Sequence[Placement]] = None,
-) -> DTensor:
+    partition_fn: Optional[Callable[[str, nn.Module], None]] = None,
```

Comment:
API discussion: should we make `partition_fn`, `input_fn` and `output_fn` kw-only arguments?

Reply:
Do you see potentially more args added to this API?

Reply:
Currently I don't see there will be newly added arguments, the function arguments should be flexible enough to do anything user want. But we should run this with a larger group and see if we need to add new things in the future.